File: spec/java/src/io/kaitai/struct/spec/TestDebugArrayUserEofException.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.DebugArrayUserEofException;
import org.testng.annotations.Test;
import static org.testng.Assert.*;
import io.kaitai.struct.KaitaiStream;
public class TestDebugArrayUserEofException extends CommonSpec {

    @Test
    public void testDebugArrayUserEofException() throws Exception {
        final DebugArrayUserEofException r = DebugArrayUserEofException.fromFile(SRC_DIR + "nav_parent_codes.bin");

        assertThrows(java.nio.BufferUnderflowException.class, new ThrowingRunnable() {
            @Override
            public void run() throws Throwable {
                // --debug implies --no-auto-read
                r._read();
            }
        });

        assertIntEquals(r.oneCat().meow(), 3);
        assertIntEquals(r.oneCat().chirp(), 73);
        assertIntEquals(r.arrayOfCats().size(), 3);
        assertIntEquals(r.arrayOfCats().get((int) 0).meow(), 49);
        assertIntEquals(r.arrayOfCats().get((int) 0).chirp(), 50);
        assertIntEquals(r.arrayOfCats().get((int) 1).meow(), 51);
        assertIntEquals(r.arrayOfCats().get((int) 1).chirp(), 66);
        assertIntEquals(r.arrayOfCats().get((int) 2).meow(), 98);
    }
}

File: spec/python/spec/test_debug_array_user_eof_exception.py
import unittest
from testformats.debug_array_user_eof_exception import DebugArrayUserEofException

class TestDebugArrayUserEofException(unittest.TestCase):
    def test_debug_array_user_eof_exception(self):
        with DebugArrayUserEofException.from_file('src/nav_parent_codes.bin') as r:
            with self.assertRaises(EOFError):
                # --debug implies --no-auto-read
                r._read()

            self.assertEqual(r.one_cat.meow, 3)
            self.assertEqual(r.one_cat.chirp, 73)
            self.assertEqual(len(r.array_of_cats), 3)
            self.assertEqual(r.array_of_cats[0].meow, 49)
            self.assertEqual(r.array_of_cats[0].chirp, 50)
            self.assertEqual(r.array_of_cats[1].meow, 51)
            self.assertEqual(r.array_of_cats[1].chirp, 66)
            self.assertEqual(r.array_of_cats[2].meow, 98)
